experimental/parser/parse_decl.go

```diff
@@ -302,12 +303,28 @@ func parseRange(p *parser, c *token.Cursor) ast.DeclRange {
 			required: true,
 			exhaust:  false,
 			parse: func(c *token.Cursor) (ast.ExprAny, bool) {
+				first = false
 				expr := parseExpr(p, c, in.In())
 				badExpr = expr.IsZero()
 
 				return expr, !expr.IsZero()
 			},
-			canStart: canStartExpr,
+			start: canStartExpr,
+			stop: func(t token.Token) bool {
+				if t.Text() == ";" || t.Text() == "[" {
+					return true
+				}
+
+				// After the first element, stop if we see an identifier
+				// coming up. This is for a case like this:
+				//
+				// reserved 1, 2
+				// message Foo {}
+				//
+				// If we don't do this, message will be interpreted as an
+				// expression.
+				return !first && t.Kind() == token.Ident
```

Comment:
This might be worth a TODO since editions uses identifiers instead of strings for reserved field names. So this means that `reserved foo, bar baz` will treat `baz` as a new decl, which is less than ideal.

Reply:
Figured out a way to resolve this w/o affecting the happy path.